src/utils/form.ts

```diff
@@ -12,3 +14,148 @@ export const hiddenInputStyle = `
   -webkit-appearance: none !important;
   z-index: -1 !important;
 `;
+
+interface FormDataEvent extends Event {
+  formData: FormData;
+}
+
+/**
+ * Defines interface for components that are form-compatible.
+ *
+ * Along with the interface, use the matching form utils to help set up the component behavior.
+ */
+export interface FormAssociated<T = any> {
+  /**
+   * The host element.
+   */
+  readonly el: HTMLElement;
+
+  /**
+   * The form this component is associated with.
+   */
+  formEl: HTMLElement;
```

Comment:
For label, i'm using `effectiveLabel` but I like this better. I'll update my interface names to be consistent with these.